### PR TITLE
fix(atomic): added cypress tests for result table and added error to table

### PR DESCRIPTION
**Pull request by btaillon-coveo**
https://coveord.atlassian.net/browse/KIT-967

### CODE REVIEW

File: packages/atomic/cypress/utils/elementUtils.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+function isElement(node: Node): node is Element {
+  return node.nodeType === 1;
+}
+
+export function getDeepText(node: Node): string {
+  if (node.nodeType === 3) {
```
Comment (ThibodeauJF) [inline packages/atomic/cypress/utils/elementUtils.ts:6]: can we add a clear isTextNode util and remove the comment?

File: packages/atomic/cypress/integration/result-list/result-table.cypress.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
+import {generateComponentHTML, TestFixture} from '../../fixtures/test-fixture';
+import {getDeepText} from '../../utils/elementUtils';
+import * as CommonAssertions from '../common-assertions';
+import {resultTextComponent} from './result-components/result-text-selectors';
+import {ResultListSelectors} from './result-list-selectors';
+import {addResultTable} from './result-table-actions';
+import {ResultTableSelectors} from './result-table-selectors';
+
+describe('Result List (Table) Component', () => {
+  it('should load', () => {
```
Comment (ThibodeauJF) [inline packages/atomic/cypress/integration/result-list/result-table.cypress.ts:10]: Thinking about these "should load" tests, all other tests kind of depend on the element loading. Do we need to have a unique setup for this, which amount to more test time, or can we put the expectation in another "describe" block?

**Review by ThibodeauJF — APPROVED** [review]

**Review by lvu285 — APPROVED** [review]

File: packages/atomic/cypress/integration/result-list/result-table.cypress.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+      ResultTableSelectors.placeholder().should('be.visible');
+    });
+  });
+
+  describe('with no columns', () => {
+    beforeEach(() => {
+      new TestFixture().with(addResultTable([])).init();
+    });
+
+    // CommonAssertions.assertConsoleError();
```
[the remark this answers is not in the document]
Comment (btaillon) [reply packages/atomic/cypress/integration/result-list/result-table.cypress.ts:39]: This doesn't pass, I forgot to fix it

File: packages/atomic/cypress/integration/result-list/result-table.cypress.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        );
+        const expectedCellContent = columns.map((col) => col.content.outerHTML);
+        expect(actualCellContent).to.deep.eq(expectedCellContent);
+      });
+    });
+  });
+
+  describe('with a column containing a result template component', () => {
+    const field = 'train';
+    const fieldValue = 'Orient Express';
```
Comment (samisayegh) [inline packages/atomic/cypress/integration/result-list/result-table.cypress.ts:70]: The train of kings, and the king of trains.

**Pull request merged**